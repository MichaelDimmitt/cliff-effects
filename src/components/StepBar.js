import React from 'react';
import { Step } from 'semantic-ui-react';

import { STEP_VALS } from '../forms/STEP_VALS';


const StepBar = ({ currentStepKey, goToStep, translations }) => {

  let cleanSteps = [];

  STEP_VALS.forEach((step, index) => {
    let newStep = { title: { content: translations[ `i_` + step.key ] }};

    newStep.active  = step.key === currentStepKey;
    newStep.onClick = function (event) { goToStep({ key: step.key }); };
    newStep.key     = index;

    cleanSteps[ index ] = newStep;
  });

  return (
    <Step.Group
      className = { `six` }
      size      = { `mini` }
      items     = { cleanSteps }
      ordered />
  );
};


export { StepBar };
