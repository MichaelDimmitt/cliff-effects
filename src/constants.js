import { CurrentIncomeStep } from './forms/CurrentIncome';
import { CurrentExpensesStep } from './forms/CurrentExpenses';
import { PredictionsStep } from './forms/Predictions';
import { HouseholdStep } from './forms/Household';
import { CurrentBenefitsStep } from './forms/CurrentBenefits';

export const STEPS = [
  {
    form: CurrentBenefitsStep,
    key:  'currentBenefits',
    time: 'current',
  },
  {
    form: HouseholdStep,
    key:  'household',
    time: 'current',
  },
  {
    form: CurrentIncomeStep,
    key:  'currentIncome',
    time: 'current',
  },
  {
    form: CurrentExpensesStep,
    key:  'currentExpenses',
    time: 'current',
  },
  {
    form: PredictionsStep,
    key:  'predictions',
    time: 'future',
  }, //,
  //  { title: 'Graphs', form: ResultsGraph }
]; // end STEPS
