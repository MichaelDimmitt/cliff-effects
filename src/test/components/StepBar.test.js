import React from 'react';
import { mount } from 'enzyme';

import StepBar from '../../components/StepBar';
import { STEPS } from '../../constants';
import { snippets } from '../helpers';

describe('<StepBar>', () => {
  const goToStep = jest.fn();
  const defaultProps = {
    currentStepKey: STEPS[ 0 ].key,
    goToStep:       goToStep,
    snippets:       snippets,
  };

  const buildWrapper = (props = {}) => {
    return mount(<StepBar
      { ...defaultProps }
      { ...props } />);
  };

  afterEach(() => {
    goToStep.mockReset();
  });

  it('renders steps with active prop', () => {
    expect(buildWrapper({ currentStepKey: STEPS[ 0 ].key }).find('Step').at(0).prop('active')).toBe(true);
    expect(buildWrapper({ currentStepKey: STEPS[ 1 ].key }).find('Step').at(1).prop('active')).toBe(true);
    expect(buildWrapper({ currentStepKey: STEPS[ 2 ].key }).find('Step').at(2).prop('active')).toBe(true);
  });

  it('calls goToStep with expected index', () => {
    const wrapper = buildWrapper();

    wrapper.find('Step').at(0).simulate('click');
    expect(goToStep).toHaveBeenLastCalledWith({ key: STEPS[ 0 ].key });

    wrapper.find('Step').at(1).simulate('click');
    expect(goToStep).toHaveBeenLastCalledWith({ key: STEPS[ 1 ].key });

    wrapper.find('Step').at(2).simulate('click');
    expect(goToStep).toHaveBeenLastCalledWith({ key: STEPS[ 2 ].key });
  });

  it('matches snapshot', () => {
    expect(buildWrapper()).toMatchSnapshot();
  });
});
