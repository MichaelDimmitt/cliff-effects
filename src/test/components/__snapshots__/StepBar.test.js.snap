// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepBar> matches snapshot 1`] = `
<StepBar
  currentStepKey="currentBenefits"
  goToStep={[Function]}
  snippets={null}
>
  <StepGroup
    className="six"
    items={
      Array [
        Object {
          "active": true,
          "key": 0,
          "onClick": [Function],
          "title": Object {
            "content": "Some translated text.",
          },
        },
        Object {
          "active": false,
          "key": 1,
          "onClick": [Function],
          "title": Object {
            "content": "Some translated text.",
          },
        },
        Object {
          "active": false,
          "key": 2,
          "onClick": [Function],
          "title": Object {
            "content": "Some translated text.",
          },
        },
        Object {
          "active": false,
          "key": 3,
          "onClick": [Function],
          "title": Object {
            "content": "Some translated text.",
          },
        },
        Object {
          "active": false,
          "key": 4,
          "onClick": [Function],
          "title": Object {
            "content": "Some translated text.",
          },
        },
      ]
    }
    ordered={true}
    size="mini"
  >
    <div
      className="ui mini ordered steps six"
    >
      <Step
        active={true}
        key="0"
        onClick={[Function]}
        title={
          Object {
            "content": "Some translated text.",
          }
        }
      >
        <a
          className="active step"
          onClick={[Function]}
        >
          <StepContent
            title={
              Object {
                "content": "Some translated text.",
              }
            }
          >
            <div
              className="content"
            >
              <StepTitle
                content="Some translated text."
              >
                <div
                  className="title"
                >
                  Some translated text.
                </div>
              </StepTitle>
            </div>
          </StepContent>
        </a>
      </Step>
      <Step
        active={false}
        key="1"
        onClick={[Function]}
        title={
          Object {
            "content": "Some translated text.",
          }
        }
      >
        <a
          className="step"
          onClick={[Function]}
        >
          <StepContent
            title={
              Object {
                "content": "Some translated text.",
              }
            }
          >
            <div
              className="content"
            >
              <StepTitle
                content="Some translated text."
              >
                <div
                  className="title"
                >
                  Some translated text.
                </div>
              </StepTitle>
            </div>
          </StepContent>
        </a>
      </Step>
      <Step
        active={false}
        key="2"
        onClick={[Function]}
        title={
          Object {
            "content": "Some translated text.",
          }
        }
      >
        <a
          className="step"
          onClick={[Function]}
        >
          <StepContent
            title={
              Object {
                "content": "Some translated text.",
              }
            }
          >
            <div
              className="content"
            >
              <StepTitle
                content="Some translated text."
              >
                <div
                  className="title"
                >
                  Some translated text.
                </div>
              </StepTitle>
            </div>
          </StepContent>
        </a>
      </Step>
      <Step
        active={false}
        key="3"
        onClick={[Function]}
        title={
          Object {
            "content": "Some translated text.",
          }
        }
      >
        <a
          className="step"
          onClick={[Function]}
        >
          <StepContent
            title={
              Object {
                "content": "Some translated text.",
              }
            }
          >
            <div
              className="content"
            >
              <StepTitle
                content="Some translated text."
              >
                <div
                  className="title"
                >
                  Some translated text.
                </div>
              </StepTitle>
            </div>
          </StepContent>
        </a>
      </Step>
      <Step
        active={false}
        key="4"
        onClick={[Function]}
        title={
          Object {
            "content": "Some translated text.",
          }
        }
      >
        <a
          className="step"
          onClick={[Function]}
        >
          <StepContent
            title={
              Object {
                "content": "Some translated text.",
              }
            }
          >
            <div
              className="content"
            >
              <StepTitle
                content="Some translated text."
              >
                <div
                  className="title"
                >
                  Some translated text.
                </div>
              </StepTitle>
            </div>
          </StepContent>
        </a>
      </Step>
    </div>
  </StepGroup>
</StepBar>
`;
